internal/model/virtualmtime.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2014 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package model
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/syncthing/syncthing/internal/db"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+type virtualMtimeRepo struct {
+	ns *db.NamespacedKV
+}
+
+type virtualMtimeRecord struct {
+	DiskMtime   time.Time
+	ActualMtime time.Time
+}
+
+func newVirtualMtimeRepo(ldb *leveldb.DB, folder string) *virtualMtimeRepo {
+	prefix := string(db.KeyTypeVirtualMtime) + folder
+
+	return &virtualMtimeRepo{
+		ns: db.NewNamespacedKV(ldb, prefix),
+	}
+}
+
+func (r *virtualMtimeRepo) UpdateMtime(path string, diskMtime, actualMtime time.Time) {
+	// TODO need error handling?
```

Comment:
I saw `MarshalBinary()` but wasn't sure if the length of the data was guaranteed to remain constant and didn't want to muck about with my own "custom" marshaling so I just threw JSON in there as a way to get off the ground quickly.  It's a (relatively) minor detail that I've no problems with changing, for sure.